lib/model/query/builder.js

```diff
@@ -10,21 +10,25 @@
 // Some helper functions that convert the plain objects defined in each querymodule
 // file in this directory into an actual query module object that will run queries.
 
-const { FutureQuery } = require('./future-query');
+const { resolve } = require('../../util/promise');
+const { postgresErrorToProblem } = require('../../util/db');
 
 // queryModuleBuilder performs the shuffling necessary to expose an interface
-// matching the definition modules but returning wrapped FutureQuerys instead of
-// plain Promises.
+// matching the definition modules but with dependencies injected automatically
+// and database errors handled.
 const queryModuleBuilder = (module, container) => {
   const result = {};
-
   for (const methodName of Object.keys(module)) {
-    result[methodName] = (...args) =>
-      new FutureQuery(module[methodName](...args), { container });
+    result[methodName] = (...args) => {
+      const result = module[methodName](...args)(container)
+      if (result.pipe != null) return resolve(result);
+      return result.catch(postgresErrorToProblem);
+    };
   }
 
+  // helper-nested functions differ in that we never assume the result is a run
+  // query, and so we do not coerce catch on it. this is important for streams.
   if (module.helper != null) {
-    result.helper = {};
```

Comment:
??

Reply:
okay this is resolved. it worked by accident. having the {} assignment is better, it is now restored.